arch/arm/core/aarch32/cortex_a_r/reset.S

```diff
@@ -46,7 +46,40 @@ GTEXT(z_arm_platform_init)
  */
 SECTION_SUBSEC_FUNC(TEXT, _reset_section, z_arm_reset)
 SECTION_SUBSEC_FUNC(TEXT, _reset_section, __start)
+#if defined(CONFIG_AARCH32_ARMV8_R)
+    /* Check if we are starting in HYP mode */
+    mrs r0, cpsr
+    and r0, r0, #MODE_MASK
+    cmp r0, #MODE_HYP
+    bne EL1_Reset_Handler
+
+    /* Init HSCTLR see Armv8-R AArch32 architecture profile */
+    ldr r0, =(HSCTLR_RES1 | SCTLR_I_BIT | SCTLR_C_BIT)
+    mcr p15, 4, r0, c1, c0, 0
+
+    /* Init HACTLR: Enable EL1 access to all IMP DEF registers */
+    ldr r0, =HACTLR_INIT
+    mcr p15, 4, r0, c1, c0, 1
+
+    /* Change EL1 exception base address */
+    ldr r0,=_vector_table
+    mcr p15, 0, r0, c12, c0, 0 /* Write to VBAR */
+
+    /* Go to SVC mode */
+    mrs r0, cpsr
+    mov r1, #MODE_SVC
+    bfi r0, r1, #0, #5
```

Comment:
A macro for "5" might make this more readable.  Another possibility to make this more readable could be
```suggestion
    bic r0, #MODE_MASK
    orr r0, #MODE_SVC
```

Reply:
Thanks picked up your suggestion.